modules/apps/server/server-admin-web/src/main/java/com/liferay/server/admin/web/internal/portlet/action/EditServerMVCActionCommand.java

```diff
@@ -295,19 +287,32 @@ private static void _resetLogLevels(
 		}
 	}
 
-	private static void _updateLogLevels(
-		Map<String, String> logLevels, String osgiServiceIdentifier) {
-
-		EditServerMVCActionCommand editServerMVCActionCommand =
-			(EditServerMVCActionCommand)
-				IdentifiableOSGiServiceUtil.getIdentifiableOSGiService(
-					osgiServiceIdentifier);
+	private static void _updateLogLevels(Map<String, String> logLevels) {
+		for (Map.Entry<String, String> logLevelEntry : logLevels.entrySet()) {
+			Log4JUtil.setLevel(
+				logLevelEntry.getKey(), logLevelEntry.getValue(), true);
+		}
 
-		if (editServerMVCActionCommand == null) {
+		if (!ClusterExecutorUtil.isEnabled()) {
 			return;
 		}
 
-		editServerMVCActionCommand._updateLogLevels(logLevels);
+		if (ClusterMasterExecutorUtil.isMaster()) {
```

Comment:
@MrJohn1911 A concern here is the possibility of creating a infinite broadcast.

Let's say this method is invoked on the master node; then it sends multicast request which is sent to all members. Then looking at one of the member, it executes this method upon receiving the cluster message, and then it will go to the `else {` block which executes this method again on master.

Looking at the usages in `FinderCacheImpl`/`EntityCacheImpl`, we always use 2 methods to do this. One method does the real logic, and the other one calls the real one, and then sends cluster request to ask other nodes to call the real one.

Reply:
Hey Dante, I just took a look at it, and I think you didn't notice that the broadcast that the master node sends its to execute the `_resetLogLevels` method, and not the `_updateLogLevels`. So the loop doesn't happen.